pkgs/development/libraries/science/chemistry/cppe/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, cmake
+, python, pybind11 ? null, numpy ? null, h5py ? null, numba ? null, scipy ? null
+, pandas ? null, pytest ? null, enablePython ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cppe";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "maxscheurer";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "guM7+ZWDJLcAUJtPkKLvC4LYSA2eBvER7cgwPZ7FxHw=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  propagatedBuildInputs = lib.optionals enablePython [
+    python
+    pybind11
+    numpy
+    h5py
+    numba
+    scipy
+    pandas
+    pytest
```

Comment:
This probably shouldn't be a ``propagatedBuildInput``, I'd expect it in ``checkInputs`` (and maybe use ``pytestCheckHook`` instead for auto-testing)

Reply:
propagating pytest is a big no no because it makes it really hard to get any other version for checkPhase.